breathe/apidoc.py

```diff
@@ -124,6 +124,20 @@ def recurse_tree(args):
                             compound.get('refid'), args)
 
 
+class TypeAction(argparse.Action):
+    def __init__(self, option_strings, dest, **kwargs):
+        super(TypeAction, self).__init__(option_strings, dest, **kwargs)
+        self.default = TYPEDICT.keys()
+        self.metavar = ','.join(TYPEDICT.keys())
```

Comment:
I think if I wasn't setting `default` and `metavar` that the `__init__` function could be left out and `argparse.Action`'s would be called instead.  I originally had this as a CsvAction to be more generic, but ran into problems with the validation logic (it liked to validate the string before it had been parsed).  I wrote this in a car looking up API's on my phone, so while it does function there may be a better implementation.  The main reason I set those vars there was since I had to make it specific to the `TYPEDICT` anyways I figured I may as well move all of the information to the class instead of scattering some in the class and some in the `parser.add_argument` call.